docs/DeveloperGuide.md

```diff
@@ -321,6 +366,14 @@ As mentioned, a task may or may not belong to a module. In the case it does not,
 The original AB3 implementation of edit commands, which would default to the original field if the edited
  field was null, would not be sufficient. Hence, we added 2 additional boolean variables - `isRemarkChanged` and
   `isCodeChanged`, to know whether users wanted to remove the existing module code or remark.
+  
+Another design challenge was how to manage our predicates. As mentioned above, `getModuleTasks`, `getOverdueTasks`, 
+`getDayUpcomingTasks`, and `getFutureTasks` all make use of different predicates. Hence, we had 2 options:
+
+|  | Pros | Cons |
+| ---- | ----- | ------- |
+| **Option 1 (current choice):** Extract each the predicates into their own unique class: <ul><li>`TaskAfterDatePredicate`</li><li>`TaskHasCodePredicate`</li><li>`TaskIsOverduePredicate`</li><li>`TaskOnDatePredicate`</li></ul> | Increases code maintainability and testability. Now as a developer you exactly where to find each predicate. Makes use of the DRY principle and improves abstraction because you no longer need to interact with the actual lambda or test function, simply call the predicate. | Makes code more verbose as each predicate can simply be declared using a single lambda. |
+| **Option2:** Declare each predicate using a single lambda in the ModelManager class. No predicate will have a class. | Makes code shorter and simpler to read. No need to create a class when you can simply declare a predicate with a lambda. | Need to duplicate such code when using ModelStubs for testing. This will violate the DRY principle. |
```

Comment:
Right now I think that this predicate section should be put in like a "general design consideration" section, because for lessons they will copy-pasted this exact thing, but on the other hand doing that will make our DG looks "shorter" and "less effort", so I'm not sure 😢What do you think?

Reply:
Hmm yea I will change it to the general overview section